src/net/connection/quality.rs

```diff
@@ -25,4 +100,31 @@ mod test {
             .unwrap();
         let _new_conn = VirtualConnection::new(addr.next().unwrap());
     }
+
+    #[test]
+    fn convert_duration_to_milliseconds_test() {
+        let network_quality = NetworkQualityMeasurer::new(NetworkConfig::default());
+        let milliseconds1 = network_quality.as_milliseconds(Duration::from_secs(1));
+        let milliseconds2 = network_quality.as_milliseconds(Duration::from_millis(1500));
+        let milliseconds3 = network_quality.as_milliseconds(Duration::from_millis(1671));
+
+        assert_eq!(milliseconds1, 1000);
+        assert_eq!(milliseconds2, 1500);
+        assert_eq!(milliseconds3, 1671);
+    }
+
+    #[test]
+    fn smooth_out_rtt()
+    {
+        let mut config = NetworkConfig::default();
+        // for test purpose make sure we set smoothing factor to 10%.
+        config.rtt_smoothing_factor = 0.10;
+        config.rtt_max_value = 250;
+
+        let network_quality = NetworkQualityMeasurer::new(config.clone());
+        let smoothed_rtt = network_quality.smooth_out_rtt(300);
+
+        // 300ms has exceeded 50ms over the max allowed rtt. So we check if or smoothing factor is now 10% from 50.
+        assert_eq!(smoothed_rtt, 5.);
```

Comment:
I assume 5. is equivalent to 5.0? If so, can we make this 5.0?

Reply:
Yes, it will, and yes I can change that.